IPython/html/static/notebook/js/notebook.js

```diff
@@ -2209,13 +2209,20 @@ define(function (require) {
         } else if (this.nbformat_minor < nbmodel.nbformat_minor) {
             this.nbformat_minor = nbmodel.nbformat_minor;
         }
-        
+
         if (this.session === null) {
             var kernel_name = utils.get_url_param('kernel_name');
             if (kernel_name) {
                 this.kernel_selector.set_kernel(kernel_name);
             } else if (this.metadata.kernelspec) {
                 this.kernel_selector.set_kernel(this.metadata.kernelspec);
+            } else if (this.metadata.language) {
+                // compat with IJulia, IHaskell, and other early kernels
+                // adopters that where seting a language metadata.
+                this.kernel_selector.set_kernel(this.metadata.language);
```

Comment:
When a single string is used, it must be the kernelspec name, not the language. Since this is the language, it should be passed as:

```javascript
set_kernel({
  name: "Unknown", // or something indicating that it's not a value that makes sense
  language: this.metadata.language,
});
```